provider/virtio_fs_utils.py

```diff
@@ -0,0 +1,232 @@
+
+    create_sub_folder_test(params, session, fs_dest, fs_source)
+
+
+def create_sub_folder_test(params, session, guest_dest, host_dir):
+    """
+    Test for creating the sub folder at the shared directory.
+
+    :param params: Dictionary with the test parameters
+    :param session: The session from guest
+    :param guest_dest: The shared directory on guest
+    :param host_dir: The shared directory on host
+    """
+    os_type = params.get("os_type")
+    folder_name = params.get("sub_folder_name", "virtio_fs_folder_test")
+    try:
+        error_context.context("Create the sub folder on shared directory "
+                              "of guest: ", LOG_JOB.info)
+        if os_type == "linux":
+            session.cmd("mkdir -p %s" %
+                        (guest_dest + "/" + folder_name + "/a"))
+        else:
+            fs_dest = guest_dest.replace("/", "\\")
+            session.cmd("md %s" % (fs_dest + "\\" + folder_name + "\\a"))
+
+        error_context.context("Check the sub folder on shared directory "
+                              "of host: ", LOG_JOB.info)
+        if os.path.exists(host_dir + "/" + folder_name + "/a"):
+            error_context.context("Find the %s on the host." %
+                                  (host_dir + "/" + folder_name + "/a"),
+                                  LOG_JOB.info)
+        else:
+            LOG_JOB.error("Do NOT find the sub folder on the host.")
+    finally:
+        error_context.context("Delete the sub folder on shared directory "
+                              "of guest: ", LOG_JOB.info)
+        if os_type == "linux":
+            session.cmd("rm -rf %s" % (guest_dest + "/" + folder_name))
+        else:
+            session.cmd("rmdir /s/q %s" % (guest_dest + "\\" + folder_name))
+
+
+def install_psexec(vm):
+    """
+    Copying psexec.exe from host to guest.( Windows only )
+
+    :param vm: the vm object
+    :return psexec_path: the psexec path in guest
+    """
+    src_file = os.path.join(data_dir.get_deps_dir("psexec"), "PsExec.exe")
+    dest_dir = "%systemdrive%\\"
+    vm.copy_files_to(src_file, dest_dir)
+    return os.path.join(dest_dir, "PsExec.exe")
+
+
+def basic_io_test_via_psexec(test, params, vm, usernm, pwd):
+    """
+    Use psexec to do the io test( Windows only ).
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param vm: the vm object
+    :param usernm: the username used to execute the cmd
+    :param pwd: the password used to execute the cmd
+    """
+    if params.get("os_type", "windows") == "windows":
+        error_context.context("Running viofs basic io test via psexec",
+                              LOG_JOB.info)
+        cmd_dd_win = params.get("virtio_fs_cmd_dd_win",
+                                "dd if=/dev/random of=%s bs=1M count=100")
+        test_file = params.get('virtio_fs_test_file', "virtio_fs_test_file")
+        io_timeout = params.get("fs_io_timeout", 120)
```

Comment:
in this way, we'll get a string io_timeout if fs_io_timeout is set in cfg, which will cause exception when passing it as an argument of cmd or run function @nickzhq